controllers/cloudstackmachine_controller.go

```diff
@@ -199,14 +194,55 @@ func (r *CloudStackMachineReconciliationRunner) GetOrCreateVMInstance() (retRes
 	return ctrl.Result{}, err
 }
 
+func (r *CloudStackMachineReconciliationRunner) getMachineZone() infrav1.CloudStackZone {
+	// Get the CloudStackZone for the Machine.
+	var machineZone infrav1.CloudStackZone
+	for _, zone := range r.Zones.Items {
+		machineZone = zone
+		if zone.Spec.ID == r.ReconciliationSubject.Status.ZoneID {
+			break
+		}
+	}
+	return machineZone
+}
+
+// AttachISOIfConfigured attach ISO to vm instance
+func (r *CloudStackMachineReconciliationRunner) StartVMInstance() (retRes ctrl.Result, err error) {
+	if len(*r.ReconciliationSubject.Spec.InstanceID) == 0 {
+		return r.RequeueWithMessage("vm instance not yet available.")
+	}
+	r.Log.Info("Start vm instance.")
+
+	err = r.CSUser.StartVMInstance(r.ReconciliationSubject)
+
+	return ctrl.Result{}, err
+}
+
+// AttachISOIfConfigured attach ISO to vm instance
+func (r *CloudStackMachineReconciliationRunner) AttachISOIfConfigured() (retRes ctrl.Result, err error) {
+	if len(r.ReconciliationSubject.Spec.ISOAttachment.ID) == 0 && len(r.ReconciliationSubject.Spec.ISOAttachment.Name) == 0 {
+		// ISO attachment not configured, ignore
+		return ctrl.Result{}, nil
+	}
+	if len(*r.ReconciliationSubject.Spec.InstanceID) == 0 {
+		return r.RequeueWithMessage("vm instance not yet available.")
+	}
+	r.Log.Info("Start to attach ISO.")
```

Comment:
might be helpful to include some identifiers here - for the ISOAttachment and the VM instance